src/pages/fieldManagement/settings/defaultFieldComponents/RelationshipEditor.jsx

```diff
@@ -0,0 +1,235 @@
+    currentRelationships.push(newRelationship);
+  allRelationships[currentCategory] = currentRelationships;
+  return allRelationships;
+}
+
+function renderSingleRelationship(
+  allRelationships,
+  currentCategory,
+  relationship,
+  onChange,
+) {
+  return (
+    <div style={{ marginLeft: 32, marginTop: 10 }} key={relationship}>
+      <TextInput
+        width={240}
+        schema={relationship}
+        onChange={newRelationshipName => {
+          onChange(
+            updateRelationshipInCategory(
+              allRelationships,
+              currentCategory,
+              relationship,
+              newRelationshipName,
+            ),
+          );
+        }}
+        value={relationship}
+        autoFocus
+        InputProps={{
+          endAdornment: (
+            <InputAdornment position="end">
+              <DeleteButton
+                onClick={() => {
+                  onChange(
+                    deleteRelationship(
+                      allRelationships,
+                      currentCategory,
+                      relationship,
+                    ),
+                  );
+                }}
+              />
+            </InputAdornment>
+          ),
+        }}
+      />
+    </div>
+  );
+}
+
+function renderSingleRelationshipObj(
+  allRelationships,
+  targetKey,
+  onChange,
+) {
+  const relationships = get(allRelationships, targetKey);
+  return (
+    <div style={{ marginTop: 10 }} key={targetKey}>
+      <TextInput
+        width={240}
+        schema={targetKey}
+        onChange={newName => {
+          onChange(
+            updateCategoryName(allRelationships, targetKey, newName),
+          );
+        }}
+        value={targetKey}
+        autoFocus
+        InputProps={{
+          endAdornment: (
+            <InputAdornment position="end">
+              <IconButton
+                size="small"
+                onClick={() => {
+                  onChange(
+                    addRelationshipToCategory(
+                      allRelationships,
+                      targetKey,
+                      '',
+                    ),
+                  );
+                }}
+              >
+                <NewChildIcon />
+              </IconButton>
+              <DeleteButton
+                onClick={() => {
+                  onChange(
+                    deleteCategory(allRelationships, targetKey),
+                  );
+                }}
+              />
+            </InputAdornment>
+          ),
+        }}
+      />
+      {relationships.map(relationship =>
+        renderSingleRelationship(
+          allRelationships,
+          targetKey,
+          relationship,
+          onChange,
+        ),
+      )}
+    </div>
+  );
+}
+
+export default function RelationshipEditor({
+  schema,
+  value,
+  onChange,
+  siteSettings, // unpacking to prevent passing to div
+  ...rest
+}) {
+  const justRelationshipsFormPart = get(
+    value,
+    'relationships',
+    value,
+  );
```

Comment:
this also seems weird to me. fallback to value if relationships.value isn't found?

Reply:
Certainly intentional. See above.